### PR TITLE
Add unpatch to dropconnect

### PR DESCRIPTION
**Pull request by Dref360**
## Summary:

This is a bit hacky as we need to keep the "old" Dropout value, but I think this does what it needs.

## Checklist:

* [x] Your code is documented (To validate this, add your module to `tests/documentation_test.py`).
* [x] Your code is tested with unit tests.
* [x] You moved your Issue to the PR state.

### CODE REVIEW

**Review by parmidaatg — CHANGES_REQUESTED** [review]
Line 58 in `test/bayesian/dropout_test` can be replaced by your new test util `is_deterministic` right?
and line 36 in `test/bayesian/dropconnect_test.py`

File: baal/bayesian/weight_drop.py [hunk 1]
```diff
@@ -88,39 +99,61 @@ def patch_module(
...
+    changed = replace_layers_in_module(
+        module, _dropconnect_mapping_fn, layers=layers, weight_dropout=weight_dropout
+    )
     if not changed:
         warnings.warn("No layer was modified by patch_module!", UserWarning)
     return module
 
 
-def _patch_layers(module: torch.nn.Module, layers: Sequence, weight_dropout: float) -> bool:
+def unpatch_module(module: torch.nn.Module, inplace: bool = True) -> torch.nn.Module:
```
[the remark this answers is not in the document]
Comment (Dref360) [reply baal/bayesian/weight_drop.py:110]: you are right, we could uniformize all of these.

**Review by parmidaatg — APPROVED** [review]
amazing :D lets go 👍🏼
